dietpi/dietpi-software

```diff
@@ -17651,6 +17653,57 @@ List of installed software and their online documentation URLs:
 			# Load all automation vars
 			FirstRun_Automation_Init
 
+			if (( $AUTOINSTALL_RESTORE )); then
+
+					# Find unmounted drives
+					lsblk --noheadings --raw -o NAME,MOUNTPOINT | mawk '$1~/s.*[0-9]/ && $2 == ""' > .lsblk_out_tmp
```

Comment:
This allows only devices of type "sda1" while "mmcblk0p1", "nvme0n0p1" and other types exist as well, and of course a file system can be flashed onto a block device without partition table, so that it would be "sda" only.

Three alternatives:

```
root@micha:~# lsblk -rnpo NAME,UUID,MOUNTPOINT | mawk '$2 && ! $3 {print $1}'
/dev/sdb
```
If it has a UUID (hence file system) but not mount point, print it's full device path (`-p` option).

```
root@micha:~# blkid -o device
/dev/mmcblk0p1
/dev/mmcblk0p2
/dev/sda
/dev/sdb
```
`blkid` prints only block devices with UUIDs/file systems. In the `read` loop `findmnt $line > /dev/null || continue` can be used to mount only the ones which are not yet mounted. What I don't like about `blkid` is that it deeply probes all attached drives, so they spin up and it may hang for a while. Not sure what info it gets from this or if it makes it more trustable than `lsblk` which does not do such spin-up probe but is able to print much more information.